ldap/servers/plugins/uiduniq/uid.c

```diff
@@ -1364,7 +1364,7 @@ preop_modrdn(Slapi_PBlock *pb)
 
     /* Apply the rename operation to the dummy entry. */
     /* slapi_entry_rename does not expect rdn normalized */
-    err = slapi_entry_rename(e, rdn, deloldrdn, superior);
+    err = slapi_entry_rename(e, rdn, deloldrdn, destinationSDN);
```

Comment:
This is the last thing that I do not understand. 
Initially it was called with 'superior'. It made sens as the fourth param is expected to be the parent of the destination entry.
But now, if SLAPI_MODRDN_NEWSUPERIOR_SDN is NULL (case an entry is just renamed not moved), then destinationSDN takes the value of sourceSDN. It should take the value of the parent of sourceDN.